### PR TITLE
feat: Adds Legend chart typings and demo

### PR DESCRIPTION
**Pull request by Golodhros**
## Description
Adds the legend component types
Adds demo and sample data

## Motivation and Context
Completing our types

## How Has This Been Tested?
Manually

## Screenshots (if appropriate):
Britecharts_test_for_typescript_

## Types of changes

- [ ] Refactor (changes the way we code something without changing its functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] Latest master code has been merged into this branch
- [ ] No commented out code (if required, place // TODO above with explanation)
- [ ] No linting issues
- [ ] Build is successful
- [ ] Updated the documentation
- [ ] Added tests to cover changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: src/typings/common/base.d.ts [hunk 1]
```diff
@@ -14,9 +14,18 @@ export interface ChartBaseAPI<T> {
...
-  width(width?: number): T & ChartBaseAPI<T>;
+  height(height?: number): T & ChartBaseAPIMinimal<T>;
+  loadingState(markup?: string): T & ChartBaseAPIMinimal<T>;
+  locale(localObject?: LocalObject | null): T & ChartBaseAPIMinimal<T>;
+  margin(margin?: ChartMarginParams): T & ChartBaseAPIMinimal<T>;
+  width(width?: number): T & ChartBaseAPIMinimal<T>;
+}
+
+
+export interface ComponentBaseAPI<T> {
```
Comment (DalerAsrorov) [inline src/typings/common/base.d.ts:25]: It makes sense if the `ComponentBase` interface extends the `ChartBaseAPIMinimal`, then we can remove the duplicate methods and have just `numberFormat` and `colorSchema` for `ComponentBaseAPI`:
```
export interface ComponentBaseAPI <T> extends ChartBaseAPIMinimal <T>  {
 //...
}
```

File: src/sandbox/legend.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    .margin({
+      left: 20,
+      right: 60,
+      top: 20,
+      bottom: 0,
+    })
+    .width(containerNode.clientWidth)
+    .height(200)
+    .colorSchema(colors.colorSchemas.britecharts)
+    // .highlight(11);
```
Comment (DalerAsrorov) [inline src/sandbox/legend.ts:16]: we can remove the commented part right?

File: src/data/legend-sample.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { LegendDataShape } from '@charts/legend-component';
+
```
Comment (DalerAsrorov) [inline src/data/legend-sample.ts:2]: the formatting seems to be off in this file - I believe the tabs should be smaller (2 spaces?).

File: src/typings/common/base.d.ts (hunk 1 above)
Comment (Golodhros) [reply src/typings/common/base.d.ts:25]: Totally, I agree.

I was thinking about having a base more componetized, but I think it makes sense for us to wait until we have all the charts, and then find the right abstraction for it.

File: src/sandbox/legend.ts (hunk 2 above)
Comment (Golodhros) [reply src/sandbox/legend.ts:16]: Ooops, forgot that while debugging. I will remove it. 
Thanks!

File: src/data/legend-sample.ts (hunk 3 above)
Comment (Golodhros) [reply src/data/legend-sample.ts:2]: True, fixing!

**Pull request merged**